lib/Target/LLVMIR/LLVMDIScope.cpp

```diff
@@ -0,0 +1,148 @@
+#include "triton/Target/LLVMIR/Passes.h"
+
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/Pass.h"
+#include "llvm/BinaryFormat/Dwarf.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/Path.h"
+
+//===----------------------------------------------------------------------===//
+// This file implements a pass to add debug info scope to LLVM operations, and
+// is inspired by the DIScopeForLLVMFuncOpPass in LLVM/MLIR. Different from the
+// DIScopeForLLVMFuncOpPass, this pass also handles inlined functions.
```

Comment:
Some assumptions are specific to triton, but we should be able to make it general.
I just reminded that there are several things we made a specialized version instead of upstreaming...
Like `multiRootTopologicalSort`, `multiRootGetSlice`.